dulwich/tests/test_porcelain.py

```diff
@@ -47,7 +48,7 @@ class PorcelainTestCase(TestCase):
     def setUp(self):
         super(TestCase, self).setUp()
         repo_dir = tempfile.mkdtemp()
-        self.addCleanup(shutil.rmtree, repo_dir)
+        self.addCleanup(rmtree_ro, repo_dir)
```

Comment:
It's needed for `test_porcelain.PushTests.test_simple` which calls a `SubprocessGitClient` when it calls `push`. 

We can change the `default_local_git_client_cls` to `LocalGitClient` for this test, but I don't think there is any harm in using `rmtree_ro` here.